### PR TITLE
lp_solve: 5.5.2.5 -> 5.5.2.11

### PR DESCRIPTION
**Pull request by smancill**
###### Motivation for this change

Bump barely updated `lp_solve` to latest release and fix build on Darwin (working on this to build `shogun` on Darwin).

###### Things done

- Use the generic builder and override build and install phases instead of using `buildCommand` and missing phases and hooks, like `fixupPhase`.
- Fix build on Darwin by using the proper build script (`ccc.osx`).
  - Add `cctools` to build inputs because `ccc.osx` uses the `libtool` command.
  - The `install_name` of the library is not set so fix it with `fixDarwinDylibNames`.
- Install headers into the expected location `$out/include/lpsolve` instead of `$out/include`. This follows what other distro do, and fixes `shogun` (the only package using `lp_solve`) that cannot find it otherwise.

---------

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [X] macOS
   - [X] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all packages that depend on this change (only `shogun`).
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 134393` at f1642920 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>lp_solve</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fetchpatch`.
Near pkgs/applications/science/math/lp_solve/default.nix:1:26:
```
  |
1 | { lib, stdenv, fetchurl, fetchpatch, cctools, fixDarwinDylibNames }:
  |                          ^
```
</li>
  </ul>
</details>

**Comment by smancill** [comment]
Applied suggested change.

File: pkgs/applications/science/math/lp_solve/default.nix [hunk 1]
```diff
@@ -1,40 +1,45 @@
-{ lib, stdenv, fetchurl }:
+{ lib, stdenv, fetchurl, cctools, fixDarwinDylibNames }:
 
 stdenv.mkDerivation rec {
 
   pname = "lp_solve";
```
Comment (SuperSandro2000) [inline pkgs/applications/science/math/lp_solve/default.nix:5]:
```suggestion
stdenv.mkDerivation rec {
  pname = "lp_solve";
```

Comment (smancill) [reply pkgs/applications/science/math/lp_solve/default.nix:5]: Ah, but it was already merged. I think it's ok to leave there for now until a future update?

**Pull request merged**